test/extended/util/framework.go

```diff
@@ -219,7 +221,20 @@ func WaitForADeployment(client kclient.ReplicationControllerInterface,
 		defer w.Stop()
 
 		for {
-			val, ok := <-w.ResultChan()
+			timeout := make(chan bool, 1)
+			go func() {
+				time.Sleep(10 * time.Second)
```

Comment:
hmmm, if the watcher did not expire, blocking on the receive will eventually succeed. It makes it no better to recompute all of the things above and issue a new watcher, right?

The problem here seems to be how we do the outer timeout, because if we block on the watcher forever, the timeout is not actually observed, agree? Is that the problem this new inner timeout is trying to solve?